lxd/instance/drivers/driver_lxc.go

```diff
@@ -1732,17 +1748,36 @@ func (d *lxc) deviceHandleMounts(mounts []deviceConfig.MountEntryItem) error {
 
 			_, err = files.Lstat(relativeTargetPath)
 			if err == nil {
-				err := d.removeMount(mount.TargetPath)
-				if err != nil {
-					return fmt.Errorf("Error unmounting the device path inside container: %s", err)
+				removeTargetFiles := false
+
+				// Check if the target path hasn't been created by LXD
+				mountConf := d.deviceVolatileGetFunc(mount.DevName)()
+				for k, v := range mountConf {
```

Comment:
why iterating a map if you're looking up by key and comparing the value?

Reply:
fixed